wordpress-mobile/WordPress-Android
# Stats error when all the calls fail

Fixes #9038 

I tried several approaches to this and I think this is the best we can do. Some of the code could probably be extracted but after the change it's consistent across the blocks and works better than before. It is too long PR because we have so many usecases and the changes are the same in all of them.

* Show a fullscreen error when loading of all blocks fails. The block contains a "Retry" button which refreshes the data. 
* Show all the blocks even if they miss data - for example Latest post summary was disappearing before
* Stop loading of the DWMY blocks if the Overview usecase fails or has no data.
--
To test:
* Turn on the airplane mode
* Go to stats
* You see the empty screen with the "Retry" button
* Turn off the airplane mode
* Click "Retry"
* The screen gets reloaded with the fresh data

To test 2:
* Open stats
* Turn on the airplane mode
* Pull to refresh
* The cached data doesn't disappear

To test 3:
* Try variations of tabs and airplane mode state - selecting date, turning AM on and off, refreshing the screen.
